regions/core/compound.py

```diff
@@ -22,8 +23,30 @@ def __init__(self, region1, operator, region2):
     def contains(self, pixcoord):
         raise NotImplementedError
 
-    def to_mask(self, mode='center'):
-        raise NotImplementedError
+    def to_mask(self, mode='center', subpixels=1):
+        mask1 = self.region1.to_mask(mode=mode, subpixels=subpixels)
+        mask2 = self.region2.to_mask(mode=mode, subpixels=subpixels)
+
+        # Common bounding box
+        bbox = BoundingBox(
+            ixmin=min(mask1.bbox.ixmin, mask2.bbox.ixmin),
+            ixmax=max(mask1.bbox.ixmax, mask2.bbox.ixmax),
+            iymin=min(mask1.bbox.iymin, mask2.bbox.iymin),
+            iymax=max(mask1.bbox.iymax, mask2.bbox.iymax))
```

Comment:
Just a sanity check - are the bbox entries guaranteed to be positive?  They ought to be; if they are not, we'll get indexing like `[-1:10]` which result in empty.

Reply:
Shouldn't such a check be added [here](https://github.com/astropy/regions/blob/master/regions/core/bounding_box.py#L56)?

Reply:
@joleroi - I forgot how regions works - is `BoundingBox` always supposed to be within the array, or can it extend beyond and some clipping is supposed to occur later when BBoxes are applied?

Could you please check the existing tests for `to_mask` and see if / how it works if a region extends beyond the image? It should then behave the same for compound regions `to_mask`, i.e. succeed or fail in the same way if possible.

Let me know if it's not clear and we'll have a look together. I would start by adding a test case for that and just see what happens.

Reply:
@cdeil Apparently, BoundingBoxes can extend over the array

```
In [1]: from regions import PixCoord, CirclePixelRegion

In [2]: reg = CirclePixelRegion(PixCoord(1,1), 5)

In [3]: reg.to_mask().bbox
Out[3]: BoundingBox(ixmin=-4, ixmax=7, iymin=-4, iymax=7)
```

I will try to make this behaviour reproducible for ``CompoundPixelRegions``

Reply:
Are there cases which are  obviously a bug or at least bad behaviour, where a user extracts a mask and a negative index is silently used for slicing, i.e. an orbitrary cutout is returned, but really an error should be raised or clipping to `i_min = ` should occur?

then you might have a bigger task on your hand here ...

Reply:
If you want to finish this up and put it in, we can, and you open a reminder issue on that question for future discussion / work.